game-app/game-core/src/main/java/games/strategy/engine/framework/I18nResourceBundle.java

```diff
@@ -0,0 +1,62 @@
+package games.strategy.engine.framework;
+
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * This is the main i18n (internationalization) class to retrieve langauge dependent outputs. The
+ * subclasses of this class should implement a singleton pattern and define the path to the resource
+ * property file(s).
+ */
+public abstract class I18nResourceBundle {
+
+  private ResourceBundle bundle;
+
+  protected I18nResourceBundle() {
+    bundle = ResourceBundle.getBundle(this.getResourcePath());
+  }
+
+  /**
+   * @param locale which should be checked whether it is supported
+   * @return True value, if it is supported, false value otherwise
+   */
+  public static boolean isSupported(final Locale locale) {
+    final Locale[] availableLocales = Locale.getAvailableLocales();
+    return Arrays.asList(availableLocales).contains(locale);
+  }
+
+  /** @return List of supported languages */
+  public static List<String> getSupportedLanguages() {
+    return Arrays.asList(Locale.ENGLISH.getLanguage(), Locale.GERMAN.getLanguage());
+  }
```

Comment:
No need to return a List of strings here, I think it's better to return `List<Locale>`, and not call `getLanguage()` at all.
This makes a difference for `en-US` and `en-GB` for example, because those are suddenly two different locales. However this would allow us to use a method like https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Locale.html#lookup(java.util.List,java.util.Collection) instead of running our own selection logic here

Reply:
What is the benefit for this change/the method `lookup` that could then be called? Currently, we only want to ensure that all supported languages (English and German in this case) are listed and I do not see how to not call `getLanguage()`. A change to `Locale` seems counterintuitive when there is a 1:n relationship from language to Locale.
 Could you explain this please a bit more?

Reply:
In the end it's all a matter of how many options you want to keep.
If you're fine with having one english (GB, US, CA etc.) translation for all regions in the world where some dialect is spoken then your approach makes perfect sense.
However when staying at a locale-based level, we could have different translations for different regions/countries, but falling back to a translation using the same language but a different region/country if no 100% correct translation is available.
So TL;DR I'd just like to keep an option for more fine grained variety of certain localisations, even if we don't plan on using them right now

Reply:
Done.